_blog/2024-02-09-using-clerk-in-rust.mdx

```diff
@@ -0,0 +1,1014 @@
+
+```tsx
+export const useGetIssues = () => {
+    const fetcher: Fetcher<IssueSchema[], string> = (url) =>
+        fetch(url).then((res) => res.json())
+    const { isLoading, data } = useSWR('/api/issues', (url) => fetcher(url));
+    return { isLoading, data };
+}
+
+```
+
+**Get issue by `id` hook**
+
+This hook uses the get issue by `id` API and returns the one single record that matches that `id` additionally, we have added a condition to call the `fetcher` function only when `id` is not empty:
+
+```tsx
+export const useGetIssue = (id: string) => {
+    const fetcher: Fetcher<{ data: IssueSchema }, string> = (url) =>
+        fetch(url).then((res) => res.json())
+    const { data } = useSWR(() => id !== "" ? `/api/issue/${id}` : null, fetcher);
+
+    return { data: data?.data }
+}
+
+```
+
+**Create issue hook**
+
+This hook returns a `trigger` method from `useSWRMutation` hook which takes the data to be sent in the request body of the create issue API and an `isMutating` boolean property which we can use to show a loading animation and disable any mutation process while this property is true.
+
+We have used the `setClose` method from the `useIssueModalStore` to close the issue modal once the issue has been created successfully
+
+The data of the payload should match the `IssueSchema` but it should not have the `id` field to it as it is a new record, so to remove `id` field we have used `Omit<IssueSchema, "id">` which creates a new type from the type provide in the first parameter and omits all those fields passed to it after the first parameter in our case it is the `id` field:
+
+```tsx
+export const useCreateIssue = () => {
+    const { setClose } = useIssueModalStore();
+
+    const create = (url: string, { arg }: { arg: Omit<IssueSchema, "id"> }) => fetch(url, {
+        method: "POST",
+        headers: {
+            Accept: "application/json",
+            "Content-Type": "application/json",
+        },
+        body: JSON.stringify(arg),
+    }).then(res => res.json())
+
+    const { mutate: revalidateIssuesList } = useSWRConfig();
+
+    const { isMutating, trigger } = useSWRMutation("/api/issue", create, {
+        onSuccess() {
+            toast.success("Issue has been created.");
+            setClose();
+            revalidateIssuesList("/api/issues")
+        },
+        onError(err) {
+            if (err.message) {
+                toast.error(err.message);
+            } else {
+                toast.error("Failed to create issue");
+            }
+        },
+    })
+
+    return { isMutating, trigger }
+}
+
+```
+
+> 🗨️ We need specify the content type header as `application/json` in the fetch calls because by default fetch API sets the content type header to `text/plain` which causes an incorrect content error as the back-end we have created expects a JSON payload.
```

Comment:
```suggestion
> 🗨️ We need to specify the content type header as `application/json` in the fetch calls because by default fetch API sets the content type header to `text/plain` which causes an incorrect content error as the back-end we have created expects a JSON payload.
```